packages/entities/entities-plugins/src/components/PluginSelectGrid.vue

```diff
@@ -0,0 +1,217 @@
+                @plugin-clicked="emitPluginData"
+              />
+            </div>
+          </template>
+
+          <div class="plugin-card-container">
+            <PluginSelectCard
+              v-for="(plugin, index) in getPluginCards(group, 'hidden')"
+              :key="index"
+              :config="config"
+              :no-route-change="noRouteChange"
+              :plugin="plugin"
+              @plugin-clicked="emitPluginData"
+            />
+          </div>
+        </KCollapse>
+      </div>
+    </template>
+  </div>
+</template>
+
+<script setup lang="ts">
+import { computed, ref, type PropType } from 'vue'
+import {
+  PluginGroup,
+  PluginGroupArray,
+  PLUGIN_GROUPS_COLLAPSE_STATUS,
+  type KongManagerPluginFormConfig,
+  type KonnectPluginFormConfig,
+  type PluginType,
+  type PluginCardList,
+  type TriggerLabels,
+} from '../types'
+import composables from '../composables'
+import PluginSelectCard from './PluginSelectCard.vue'
+
+const props = defineProps({
+  /** The base konnect or kongManger config. Pass additional config props in the shared entity component as needed. */
+  config: {
+    type: Object as PropType<KonnectPluginFormConfig | KongManagerPluginFormConfig>,
+    required: true,
+    validator: (config: KonnectPluginFormConfig | KongManagerPluginFormConfig): boolean => {
+      if (!config || !['konnect', 'kongManager'].includes(config?.app)) return false
+      if (!config.getCreateRoute) return false
+      return true
+    },
+  },
+  /**
+   * Plugins to display in the grid
+   */
+  pluginList: {
+    type: Object as PropType<PluginCardList>,
+    default: () => ({}),
+  },
+  /**
+   * @param {boolean} noRouteChange if true, let consuming component handle event when clicking on a plugin
+   * Used in conjunction with `@plugin-clicked` event
+   */
+  noRouteChange: {
+    type: Boolean,
+    default: false,
+  },
+  /**
+   * Number of plugins to always have visible (never will be collapsed)
+   */
+  pluginsPerRow: {
+    type: Number,
+    default: 4,
+  },
+})
+
+const emit = defineEmits<{
+  (e: 'plugin-clicked', plugin: PluginType): void
+}>()
+
+const { i18n: { t } } = composables.useI18n()
+const shouldCollapsed = ref<Record<string, boolean>>(PLUGIN_GROUPS_COLLAPSE_STATUS)
+
+const emitPluginData = (plugin: PluginType) => {
+  emit('plugin-clicked', plugin)
+}
+
+// remove custom plugin from original pluginList
+const nonCustomPlugins = computed((): PluginCardList => {
+  const kongPlugins = JSON.parse(JSON.stringify(props.pluginList))
+
+  delete kongPlugins[PluginGroup.CUSTOM_PLUGINS]
+
+  return kongPlugins
+})
+
+const getPluginCards = (group: string, type: 'all' | 'visible' | 'hidden') => {
+  const plugins = nonCustomPlugins.value[group as keyof PluginCardList] || []
+
+  if (type === 'all') {
+    return plugins
+  } else if (type === 'visible') {
+    return plugins.slice(0, props.pluginsPerRow)
+  }
+
+  return plugins.slice(props.pluginsPerRow)
+}
```

Comment:
This function looks identical to the `getPluginCards` implementation in the other file; could it be extracted to a composable in this package and shared?